runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -123,8 +157,15 @@ public void serialize(ByteBuf b) {
     @Override
     public void setGlobalAddress(long address) {
         super.setGlobalAddress(address);
-        this.getEntryMap().values().forEach(x -> {
-            x.setGlobalAddress(address);
-        });
+        streamUpdates.values().forEach(x -> x.setGlobalAddress(address));
+    }
+
+    public Map<UUID, MultiSMREntry> getEntryMap() {
```

Comment:
Javadoc?

Reply:
Can we add a warning on the risk of using this API?

Reply:
Done.